registry/extension/oras/artifactservice.go

```diff
@@ -116,57 +129,7 @@ func (h *referrersHandler) Referrers(ctx context.Context, revision digest.Digest
 	referrersSorted = append(referrersSorted, referrersUnsorted...)
 	return referrersSorted, nil
 }
-func (h *referrersHandler) enumerateReferrerLinks(ctx context.Context, rootPath string, ingestor func(digest.Digest) error) error {
-	blobStatter := h.extContext.Registry.BlobStatter()
-
-	return h.storageDriver.Walk(ctx, rootPath, func(fileInfo driver.FileInfo) error {
-		// exit early if directory...
-		if fileInfo.IsDir() {
-			return nil
-		}
-		filePath := fileInfo.Path()
-
-		// check if it's a link
-		_, fileName := path.Split(filePath)
-		if fileName != "link" {
-			return nil
-		}
-
-		// read the digest found in link
-		digest, err := h.readlink(ctx, filePath)
-		if err != nil {
-			return err
-		}
-
-		// ensure this conforms to the linkPathFns
-		_, err = blobStatter.Stat(ctx, digest)
-		if err != nil {
-			// we expect this error to occur so we move on
-			if err == distribution.ErrBlobUnknown {
-				return nil
-			}
-			return err
-		}
-
-		err = ingestor(digest)
-		if err != nil {
-			return err
-		}
-
-		return nil
-	})
-}
-
-func (h *referrersHandler) readlink(ctx context.Context, path string) (digest.Digest, error) {
-	content, err := h.storageDriver.GetContent(ctx, path)
-	if err != nil {
-		return "", err
-	}
-
-	linked, err := digest.Parse(string(content))
-	if err != nil {
-		return "", err
-	}
 
-	return linked, nil
+func referrersLinkPath(name string) string {
+	return path.Join("/docker/registry/", "v2", "repositories", name, "_refs", "subjects")
```

Comment:
This should be defined in the paths package

Reply:
the paths package is in the storage class and I think we decided not to add any extension specific code to the storage. And currently all of its functions and structs are not exported.